tests/baselines/reference/intersectionArithmetic.types

```diff
@@ -0,0 +1,72 @@
+=== tests/cases/conformance/types/intersection/intersectionArithmetic.ts ===
+let numberObject: number & { foo: any };
+>numberObject : number & { foo: any; }
+>foo : any
+
+let stringObject: string & { foo: any };
+>stringObject : string & { foo: any; }
+>foo : any
+
+let numberString: number & string;
+>numberString : number & string
+
+let a = numberObject + 1;
+>a : number
+>numberObject + 1 : number
+>numberObject : number & { foo: any; }
+>1 : number
+
+let b = stringObject + 1;
+>b : string
+>stringObject + 1 : string
+>stringObject : string & { foo: any; }
+>1 : number
+
+let c = numberObject + '';
+>c : string
+>numberObject + '' : string
+>numberObject : number & { foo: any; }
+>'' : string
+
+let d = stringObject + '';
+>d : string
+>stringObject + '' : string
+>stringObject : string & { foo: any; }
+>'' : string
+
+let e = numberObject + stringObject;
+>e : string
+>numberObject + stringObject : string
+>numberObject : number & { foo: any; }
+>stringObject : string & { foo: any; }
+
+let f = numberString + 1;
+>f : number
+>numberString + 1 : number
+>numberString : number & string
+>1 : number
+
+let g = numberString + '';
+>g : string
+>numberString + '' : string
+>numberString : number & string
+>'' : string
+
+let h = numberString + numberString;
+>h : number

```

Comment:
I don't know what the best behavior is here, but given that there is no way in JavaScript to create a type that is both a string and a number, it doesn't matter very much in my opinion. I think the question we should ask is: which of these rules should be used?

1. Result of `+` is `string` if at least one operand is a `string`, otherwise `number`
2. Result of `+` is `number` if both operands are `number`, otherwise `string`

In case 1, the result should be `string`, in case 2 `number`